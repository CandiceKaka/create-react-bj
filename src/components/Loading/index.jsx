//react-router切换中间过渡动画组件
import React from "react";
import styles from "./sass/index.scss";

const Loading = props => {
    if (props.error){
        console.error(props.error);
        return '报错了';
    }
    return (
        <div className={styles.container}>
            <div className={styles.inner}>
                <div className={styles.boxLoading} />
                <div className={styles.title}>努力加载中...</div>
            </div>
        </div>
    );
};
export default Loading;
