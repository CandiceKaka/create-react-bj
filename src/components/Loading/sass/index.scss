.container {
	height: 100%;
	display: flex;
	justify-content: center;
	align-items: center;
	$width: 50px;
	.inner {
		display: flex;
		align-items: center;
		flex-direction: column;
		.boxLoading {
			&:before {
				content: "";
				display: block;
				width: $width;
				height: $width;
				background: #1a6844;
				animation: animate 0.5s linear infinite;
				top: 0;
				left: 0;
				border-radius: 3px;
			}
			&:after {
				content: "";
				display: block;
				width: $width;
				height: 10px;
				background: #000;
				opacity: 0.1;
				top: $width + 9px;
				left: 0;
				border-radius: 50%;
				animation: shadow 0.5s linear infinite;
			}
		}
		.title {
			color: #666;
			font-size: 16px;
			margin-top: 14px;
		}

		@keyframes animate {
			17% {
				border-bottom-right-radius: 3px;
			}
			25% {
				transform: translateY(9px) rotate(22.5deg);
			}
			50% {
				transform: translateY(18px) scale(1, 0.9) rotate(45deg);
				border-bottom-right-radius: 40px;
			}
			75% {
				transform: translateY(9px) rotate(67.5deg);
			}
			100% {
				transform: translateY(0) rotate(90deg);
			}
		}

		@keyframes shadow {
			0%,
			100% {
				transform: scale(1, 1);
			}
			50% {
				transform: scale(1.2, 1);
			}
		}
	}
}
