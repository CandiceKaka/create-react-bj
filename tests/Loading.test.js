import React from "react";
import {shallow, mount} from "enzyme";
import renderer from "react-test-renderer";
import Enzyme from "./Enzyme";
import Loading from "../src/components/Loading";

test("component unit test", () => {
	const component = renderer.create(<Loading />);
	//快照测试
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
	//props error存在的情况的下组件渲染的应该是   '报错了'
	const ErrorLoading = shallow(<Loading error={true} />);
	expect(ErrorLoading.text()).toEqual("报错了");
	//未定义props--error的情况
	const NormalLoading = shallow(<Loading />);
	expect(NormalLoading.text()).toEqual("努力加载中...");
	//enzyme ShallowWrapper(shallow方法返回值) API

	expect(NormalLoading.find("div")).toHaveLength(4);
	//enzyme Full Rendering API
	const FullRenderLoading = mount(<Loading error={true} />);
	expect(FullRenderLoading.props()).toHaveProperty("error", true);
});
