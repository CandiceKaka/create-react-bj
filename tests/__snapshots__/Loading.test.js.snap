// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component unit test 1`] = `
<div
  className="container"
>
  <div
    className="inner"
  >
    <div
      className="boxLoading"
    />
    <div
      className="title"
    >
      努力加载中...
    </div>
  </div>
</div>
`;
